components/director/internal/repo/exist.go

```diff
@@ -41,7 +41,7 @@ func (g *universalExistQuerier) Exists(ctx context.Context, tenant string, condi
 	if tenant == "" {
 		return false, apperrors.NewTenantRequiredError()
 	}
-	conditions = append(Conditions{NewEqualCondition(*g.tenantColumn, tenant)}, conditions...)
+	conditions = append(Conditions{NewTenantIsolationCondition(*g.tenantColumn, tenant)}, conditions...)
```

Comment:
Baking the tenant isolation logic in the repo layer seems reasonable, but I have concerns.. Most of our queries and mutations do more than one db call. Take for example `appsForRuntime`, we make 3 db calls (runtime exists, getScenarios and list by scenario), this will result in executing the recursive sql 3 times on `appsForRuntime`.  I would suggest creating a separate middleware that executes the recursive query once and stores the tenant ids in the context, so that we could later add `tenant_id in (?,?,...,?)` conditions in the repo layer.

Reply:
Discussed offline and decided that there are optimizations coming (dataloaders) that will make the performance implications of this change irrelevant. However, it is much better from design PoV to leave it in the repo layer as if we move it to a central place the repository layer is not self-contained and cannot be used without the said middleware.